sourcecode/src-3/sourcecode/Macros.scala

```diff
@@ -189,8 +189,14 @@ object Macros {
         owner match {
           case defSym if defSym.isDefDef =>
             defSym.tree.asInstanceOf[DefDef].paramss
+              // FIXME Could be a List[TypeDef] too, although I'm not
+              // sure under which conditions this can happen…
```

Comment:
It can currently happen with extension methods with a type parameter clause on the extension block and the method itself (cf https://github.com/lampepfl/dotty/pull/10940), and in the (post-3.0) future we plan to allow users to add type parameter clauses after term parameter clauses in all methods.